src/Interpreters/MergeJoin.cpp

```diff
@@ -884,7 +891,7 @@ bool MergeJoin::leftJoin(MergeJoinCursor & left_cursor, const Block & left_block
             {
                 right_cursor.nextN(range.right_length);
                 right_block_info.skip = right_cursor.position();
-                left_cursor.nextN(range.left_length);
+                left_key_tail = range.left_length;
```

Comment:
Could not update the left_cursor yet, because,  maybe there are still equal keys in the right side.  Forwarding the left_cursor right now, will cause the loss of result.

Reply:
@vdimir